src/sort.c

```diff
@@ -308,7 +308,7 @@ void sortCommandGeneric(client *c, int readonly) {
      * The other types (list, sorted set) will retain their native order
      * even if no sort order is requested, so they remain stable across
      * scripting and replication. */
-    if (dontsort && sortval->type == OBJ_SET && (storekey || c->flags & CLIENT_SCRIPT)) {
+    if (dontsort && sortval->type == OBJ_SET && (storekey || c->script)) {
```

Comment:
```suggestion
    if (dontsort && sortval->type == OBJ_SET && (storekey || c->bitflags.script)) {
```
The refactor that removes the "flag" makes many of these variables quite a bit harder to read, since c->script seems more likely to indicate the script it's running than if it's a script client.

Reply:
I don't like `c->bitflags.script`. The `bitflags` part doesn't bring additional value and makes the code hard to write/read IMO. Can we consider `c->is_script_client` instead?

Reply:
> I don't like c->bitflags.script. The bitflags part doesn't bring additional value and makes the code hard to write/read IMO. Can we consider c->is_script_client instead?

I think is_script_client **only** makes more sense if that is the only accessor method, but that isn't the case. bitflags makes it clear it is a wrapper structure around a bit of lower level flags. (If space wasn't a concern bitflags.is_script_client also seems ok to me).

Reply:
It's an anonymous union, nice!

I can agree with Ping that `bitflags` (as in `c->bitflags.script`) looks a bit ugly. It exposes too much about the internal representation. Can we find another name?

`c->flags` is all flags, but with the bitfield we access one flag at a time, so how about something singular or boolean-like....

* `c->flag.script`
* `c->flagged_as.script`

Reply:
@PingXie Can you chime in here, I don't have a strong opinion and am happy to defer to whatever you think is clear.

Reply:
>c->flag.script

I like this one. 

Originally I was hoping to just go with an anonymous struct but now I notice that we actually need a named struct in places like `acceptCommonHandler`.

I think we are aligned directionally. @artikell , do you mind updating `bitflags` to `flags`? I will work with you to get this PR into Valkey 8.0

Reply:
Sure, I think we will come this week to complete these processes.